scripts/lerna-version-ci.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Get the last tagged sha from the output of "git describe"
+LAST_TAG=$(git describe --always --first-parent --abbrev=0)
+
+echo "Last Tag $LAST_TAG"
+LAST_TAGGED_SHA=$(git rev-list -n 1 $LAST_TAG)
+
+if [ -z "$LAST_TAGGED_SHA" ]; then
+  echo "Unable to find last tagged sha"
+  exit 1
+fi
+
+echo "Last tagged sha: $LAST_TAGGED_SHA"
+
+function parse_arguments() {
+  while (( "$#" )); do
+    case "$1" in
+      --all-packages)
+        AFFECTED_PROJECTS=$(yarn nx show projects | paste -sd "," -)
+        echo "Selecting all projects for version increment"
+        shift
+        ;;
+      --version-increment-type=*)
+        VERSION_INCREMENT_TYPE="${1#*=}"
+        shift
+        ;;
+      --version-increment-type*)
+        VERSION_INCREMENT_TYPE="${2}"
+        shift 2
+        ;;
+      *)
+        echo "error: unrecognized argument(s): $@"
+        exit 1
+        ;;
+    esac
+  done
+}
+
+AFFECTED_PROJECTS=""
+VERSION_INCREMENT_TYPE="patch"
+
+parse_arguments "$@"
+
+if [ -z "$AFFECTED_PROJECTS" ]; then
+  AFFECTED_PROJECTS=$(yarn nx print-affected --base $LAST_TAGGED_SHA --select=projects)
+fi
+
+echo "Affected projects: $AFFECTED_PROJECTS"
+
+# exit if no affected projects
+if [ -z "$AFFECTED_PROJECTS" ]; then
+  echo "No affected projects found. Exiting."
+  exit 1
```

Comment:
If there are no affected projects - do we want this to crash the github action?

Reply:
I think so, because if it silently does nothing you'll think you released something when you didn't. There's no downside I can see to reporting an error here.

Reply:
As long as we're consistent on the no-op = abort for other releases across the other sdks

Reply:
I don't think the same thing applies for the other non-monorepo SDKs. Because there's no need to select a subset of projects to push updates for, the idea of pushing a no-op release is reasonable. With this flow, if you want to push a dummy update you can select the "Force update all packages" option.

Reply:
I was more thinking about the C# SDK - which currently allows you to versionbump+release a no-op

Reply:
But with the force update - I think that covers it !